Spring-2019/stacks_queues/SOLUTIONS_2019Spring.md

```diff
@@ -86,20 +108,41 @@ TODO :bug:
 
 TODO :bug:
 
-## 3. PROBLEM 3 TODO :bug:
+## 3. PROBLEM 3 Validate Stack Sequences
 
-Source: TODO :bug:
+Source: [LeetCode](https://leetcode.com/problems/validate-stack-sequences/)
 
 #### Naive/Simple Solution 
 
-TODO :bug:
+For each value, push it to the stack.
 
-#### Optimal Solution
+Then, greedily pop values from the stack if they are the next values to pop.
 
-TODO :bug:
+At the end, we check if we have popped all the values successfully.
 
-#### Testing The Solutions OR Driver For Solution
+```java
+public boolean validateStackSequences(int[] pushed, int[] popped) {
+    int N = pushed.length;
+    Stack<Integer> stack = new Stack();
```

Comment:
Explicit type specification prevents an `-Xlint:unchecked` warning; seems like the thing to do